src/components/navigation-bar/types.ts

```diff
@@ -0,0 +1,20 @@
+import { ReactNode } from "react";
+
+export type NavigationItem<T = unknown> = T & {
+  id: string;
+  label: ReactNode;
+  isSelected?: boolean;
+  disabled?: boolean;
+  children?: Array<NavigationItem<T>>;
+};
+
+export type NavigationBarProps = {
+  items: NavigationItem[];
+  onItemClick?: (item: NavigationItem) => void;
+  onLogin?: () => void;
+  isAuth?: boolean;
```

Comment:
we should add opt-out feature like noLogin?: boolean that will remove auth item at all.